src/formatDate.js

```diff
@@ -1,14 +1,38 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        receivedDate['YY'] = dateArr[i].slice(2);
+        receivedDate['YYYY'] = dateArr[i];
+        break;
+      case 'YY':
+        receivedDate['YYYY'] = `${dateArr[i] < 30 ? 20 : 19}${dateArr[i]}`;
```

Comment:
The comparison `dateArr[i] < 30` is directly comparing strings, which can lead to incorrect behavior as string comparison is based on lexicographical order, not numerical. You need to convert `dateArr[i]` to a number before comparing it to 30.